### PR TITLE
Cwant/issue 103 capybara tests

### LINKED ISSUE #103: Test suite failure: upstream view requirements
There are failing view tests that are expecting content related to the upstream avalon on our pages (e.g., the avalon version, link to avalon home page, etc).

The seven failing tests are:

rspec ./spec/features/capybara_homepage_spec.rb:18 # homepage validates presence of header and footer on homepage
rspec ./spec/features/capybara_homepage_spec.rb:49 # checks navigation to external links checks navigation to Avalon Website
rspec ./spec/features/capybara_homepage_spec.rb:55 # checks navigation to external links checks navigation to Contact us page
rspec ./spec/features/capybara_homepage_spec.rb:81 # Sign in page validates presence of items on login page
rspec ./spec/features/capybara_navigation_spec.rb:18 # checks navigation after logging in checks navigation to Browse
rspec ./spec/features/capybara_navigation_spec.rb:72 # Search is able to enter keyword and perform search
rspec ./spec/features/capybara_navigation_spec.rb:78 # Search gives appropriate error when keyword returns no results

A decision needs to be made whether we try to comply with the upstream test requirements, or if we remove/replace these tests.

This issue is a refinement of issue #25.

### PR DESCRIPTION
**Pull request by cwant**
Fixes #103 

Updating the source and tests to satisfy some of the capybara-based tests. See original issue for some discussion.

### CODE REVIEW

File: app/assets/stylesheets/era_custom.css.scss [hunk 1]
```diff
@@ -488,6 +488,19 @@ body.home h4{
 	color: $link1Hover;
 }
 
+.avalon>a{
```
Comment (murny) [inline app/assets/stylesheets/era_custom.css.scss]: This file is SASS friendly, lets use SASS?

You can rewrite the following as this instead:

```
.avalon > a {
  color: $link1;
  &:visited {
    color: $link1Visited;
  }
  &:hover, &:visited:hover {
    color: $link1Hover;
  }
}
```

Also, why the child selector? Is this needed? And do you want every `.avalon` class anchor tag to have this behaviour? Or is this just for the footer?

File: app/views/modules/_footer.html.erb [hunk 2]
```diff
@@ -43,6 +43,10 @@ Unless required by applicable law or agreed to in writing, software distributed
 	</div>
         <div class="row copyright">&copy; <a href="https://library.ualberta.ca">University of Alberta Libraries</a> | University of Alberta, Edmonton, AB, Canada T6G 2R3 | <a href="https://library.ualberta.ca/about-us/policies/privacy-policy">Web Site Privacy Policy</a>
 	</div>
+    <div class="row avalon">
+      <a href="http://www.avalonmediasystem.org">Powered by Avalon</a>
+      <!-- Avalon Media System Release <%= Avalon::VERSION %> -->
+	  </div>
```
Comment (murny) [inline app/views/modules/_footer.html.erb]: Weird indentations here on the div dags?

File: app/views/modules/_header.html.erb [hunk 3]
```diff
@@ -26,6 +26,7 @@ Unless required by applicable law or agreed to in writing, software distributed
     <div class="col-sm-4">
       <form accept-charset="UTF-8" action="/catalog" id="search-form-header" method="get" role="search" class="form-inline hidden-xs mast-sm-search">
         <div style="display:none"><input name="utf8" type="hidden" value="✓"></div>
+        <input id="search_field" name="search_field" type="hidden" value="all_fields">
```
Comment (murny) [inline app/views/modules/_header.html.erb]: Why is this needed? Shouldn't this be the default of searching all the fields? What happens if a user doesn't pass in this query param? Does search break?

File: spec/features/capybara_homepage_spec.rb [hunk 4]
```diff
@@ -16,15 +16,18 @@
 describe 'homepage' do
   after { Warden.test_reset! }
   it 'validates presence of header and footer on homepage' do
+    # The theme at U of A doesn't match the upstream theme
+
     visit 'http://0.0.0.0:3000'
-    page.should have_content('Sample Content')
+    # Not relevant: page.should have_content('Sample Content')
```
Comment (murny) [inline spec/features/capybara_homepage_spec.rb]: If not needed, I would remove instead of commenting out. But guess this for for upstream or something?

File: spec/features/capybara_homepage_spec.rb [hunk 5]
```diff
@@ -48,21 +51,17 @@
 describe 'checks navigation to external links' do
   it 'checks navigation to Avalon Website' do
     visit '/'
-    click_link('Avalon Media System Project Website')
-    expect(page.status_code).to eq(200)
-    expect(page.current_url).to eq('http://www.avalonmediasystem.org/')
+    page.should have_link('Powered by Avalon',
```
Comment (murny) [inline spec/features/capybara_homepage_spec.rb]: :+1: Much better haha

File: app/views/modules/_header.html.erb (hunk 3 above)
Comment (cwant) [inline app/views/modules/_header.html.erb]: The thing runs fine if the tag is omitted, but this test fails:

https://github.com/ualbertalib/avalon/blob/master/spec/features/capybara_navigation_spec.rb#L76

I had a choice between changing the code or changing the test, so I chose to change the code to match upstream.

The code for the search box was originally here, but was moved in the local version to put it in the header:

https://github.com/ualbertalib/avalon/blob/master/app/views/modules/_avalon_search_form.html.erb#L21

File: spec/features/capybara_homepage_spec.rb (hunk 4 above)
Comment (cwant) [inline spec/features/capybara_homepage_spec.rb]: Yeah, you're right. I guess I wanted to document that we are doing something different from upstream, but this is annoying.

File: app/assets/stylesheets/era_custom.css.scss (hunk 1 above)
Comment (cwant) [inline app/assets/stylesheets/era_custom.css.scss]: Just for the footer. I should qualify that in the stylesheet. The styles immediately preceding this one are for a copyright class that is styled the same. Since that class isn't appropriate for the avalon link, I created a new style.

I have to confess that my SASS skills are minimal, but what you posted above looks a lot cleaner/DRYer. Will update.

File: app/views/modules/_footer.html.erb (hunk 2 above)
Comment (cwant) [inline app/views/modules/_footer.html.erb]: Yes, the view in avalon is a sorry mess of bad indentation (including mixed tabs/spaces in some spots). Will fix.

Comment (cwant) [inline app/views/modules/_footer.html.erb]: Now that I look at it, I should also I18n-ize the "Powered by Avalon".

File: app/views/modules/_header.html.erb (hunk 3 above)
Comment (murny) [inline app/views/modules/_header.html.erb]: Sounds like the test is broken then? Or at least its not testing anything useful. I like small URLS. Rather not just add a bunch of query params if there is zero reason for them.

But up to you.
¯\_(ツ)_/¯

File: app/assets/stylesheets/era_custom.css.scss (hunk 1 above)
Comment (murny) [inline app/assets/stylesheets/era_custom.css.scss]: Yeah theres so much CSS work that could be done here from looking at this file. Lots of repeated/duplication of classes and styles.

`.copyright` and `.avalon` ones can be merged...maybe do it to a `.footer a` class instead? I highly doubt you need the child selector as well.

Anyways lots of improvements, depends how much you want to chew off. But yes...take advantage of SASS. Maybe read some quick tutorials? Highly useful and much more readable I'd say.

File: app/assets/stylesheets/era_custom.css.scss [hunk 6]
```diff
@@ -475,17 +475,15 @@ body.home h4{
...
-.copyright>a:visited{
-	color: $link1Visited;
-}
-.copyright>a:hover{
-	color: $link1Hover;
-}
-.copyright>a:visited:hover{
-	color: $link1Hover;
+
+.footer a {
```
Comment (murny) [inline app/assets/stylesheets/era_custom.css.scss]: 👍
